mifospay/src/main/java/org/mifos/mobilewallet/mifospay/SplashScreenActivity.java

```diff
@@ -0,0 +1,16 @@
+package org.mifos.mobilewallet.mifospay;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import org.mifos.mobilewallet.mifospay.auth.ui.LoginActivity;
+
+public class SplashScreenActivity extends AppCompatActivity {
```

Comment:
Splash screens are meant to engage users by the time app is launched and prevent showing a white screen in older devices while the OS starts the first activity and it should be very quick. If we don’t want to force our app users to wait when there is nothing to load -which is not recommended doing it, then there is no need to create a different activity for the `Splash Screen`. We can implement it with a lesser line of code. Please take a look at this [PR](https://github.com/apache/fineract-cn-mobile/pull/102) or go through this [article](https://android.jlelse.eu/launch-screen-in-android-the-right-way-aca7e8c31f52).

Reply:
> Splash screens are meant to engage users by the time app is launched and prevent showing a white screen in older devices while the OS starts the first activity and it should be very quick. If we don’t want to force our app users to wait when there is nothing to load -which is not recommended doing it, then there is no need to create a different activity for the `Splash Screen`. We can implement it with a lesser line of code. Please take a look at this [PR](https://github.com/apache/fineract-cn-mobile/pull/102) or go through this [article](https://android.jlelse.eu/launch-screen-in-android-the-right-way-aca7e8c31f52).

Thanks you very much for the suggestion, I did read this article before. The current approach does handle the ColdStart's white screen but after going through other articles also, most of them suggested to use a separate activity. It does increase the code but it, in my opinion, this increases the code clarity. Thanks, if needed I shall update the PR :)

Reply:
Ook, that is nice :). Actually I don't know why most of them suggest a separate activity if we don't need any background processing, although, activities are heavy and expensive operation for OS and it should be used only if we need an independent screen, as you know even `Google`'s recommended approach for app development is a `single activity` approach. So if you think it really necessary to create an entirely different activity for showing a logo to the screen because of increasing the code clarity then that is a valid point too :)